daemon/commit.go

```diff
@@ -70,6 +74,21 @@ func (daemon *Daemon) Commit(container *Container, repository, tag, comment, aut
 	}
 	defer rwTar.Close()
 
+	blobStore := daemon.graph.BlobStore()
+	diffWriter, err := blobStore.NewWriter(crypto.SHA256)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get new writer for rootfs diff: %s", err)
+	}
+	defer func() {
+		if err != nil {
```

Comment:
quoting the golang ref/spec:

> Redeclaration does not introduce a new variable; it just assigns a new value to the original.

So this `err` is the same `err` from the result section of the function signature, right?